src/declarations/tlaDeclarations.ts

```diff
@@ -6,7 +6,35 @@ function symbolLocations(document: vscode.TextDocument, docInfo: TlaDocumentInfo
     if (!range) {
         return undefined;
     }
-    const rawWord = document.lineAt(position.line).text.substring(range.start.character, range.end.character);
+
+    // Check if the word is preceded by a dot, which would indicate it's a record field access
+    const line = document.lineAt(position.line).text;
+    if (range.start.character > 0 && line.charAt(range.start.character - 1) === '.') {
+        // This is a record field access (e.g., state.x), not a symbol reference
+        return undefined;
+    }
+
+    // Check if the word is preceded by "!." which would indicate it's in an EXCEPT expression
+    if (range.start.character > 1 && line.substring(range.start.character - 2, range.start.character) === '!.') {
+        // This is a record field in EXCEPT expression (e.g., [state EXCEPT !.x = ...])
+        return undefined;
+    }
+
+    // Check if this is a field name in a record definition (e.g., [bar |-> value])
```

Comment:
Same question as in https://github.com/tlaplus/vscode-tlaplus/pull/382#discussion_r2129524036

Reply:
was actually working on this case